### PR TITLE
inject DaoKillSwitch

### PR DESCRIPTION
**Pull request by christophsturm**
just a small refactoring to reduce global state, while i investigate transaction broadcasting.

### CODE REVIEW

File: core/src/main/java/bisq/core/dao/DaoKillSwitch.java [hunk 1]
```diff
@@ -70,8 +70,8 @@ private void applyFilter(@Nullable Filter filter) {
         daoDisabled = requireUpdateToNewVersion || filter.isDisableDao();
     }
 
-    public static void assertDaoIsNotDisabled() {
-        if (INSTANCE.isDaoDisabled()) {
```
Comment (mrosseel) [inline core/src/main/java/bisq/core/dao/DaoKillSwitch.java:74]: is the INSTANCE still needed after this refactoring?

Comment (christophsturm) [reply core/src/main/java/bisq/core/dao/DaoKillSwitch.java:74]: no.

**Comment by mrosseel** [comment]
did the simplify commit accidentally get added to this PR? Seems out of scope. Otherwise utACK

**Comment by christophsturm** [comment]
i added it because its a local change that i made and that seemed simple enough. i can put it into a different pr where it fits better

**Comment by christophsturm** [comment]
removed the unrelated commit

**Review by sqrrm — APPROVED** [review]
utACK

**Pull request merged**